### PR TITLE
refactor: consolidate kubernetes interactions into one file

### PR DESCRIPTION
**Pull request by Jackmin801**
**Goals:**

- [x] Make kubernetes tests easier to work with by consolidating kubernetes api interactions
- [x] Pull out linkerd where possible
- [x] Pull out kubernetes python client where possible
- [x] Move stuff that hasn't been ported into a legacy folder for now

Changes
- Directories used for building images have been moved from /tests/k8s to /tests/k8s/images
- Version numbers for built images have been removed. They are now all tagged with test-pip
- `docker_image` fixture has been removed in favor of directly using the image name
- The k8s tests no longer require an install of linkerd
- `create_all_flow_deployments_and_wait_ready` has been replaced with `k8s_cluster.deploy_from_dir`
- `portforward.forward` has been replaced with `k8s_cluster.port_forward`
- `core_client.list_namespaced_pod(...).items[0].metadata.name` has been replaced with `k8s_cluster.get_pod_name(...)`
- Non-blocking `core_client.delete_namespace` has been replaced with the blocking `k8s_cluster.delete_namespace`
- General try excepts around each test case have been taken out (Pytest already logs errors?)
- test_k8s_failure, test_graceful_request_handling and test_flow_monitoring have been moved to /tests/k8s_legacy which uses the old structure
- Open telemetry k8s tests have been moved into /tests/k8s/otel

### CODE REVIEW

File: tests/k8s/test_k8s.py [hunk 1]
```diff
@@ -256,566 +142,278 @@ def k8s_flow_with_needs(docker_images):
...
+    args = set_deployment_parser().parse_args(['--uses', 'DocReplaceExecutor'])
+    with Deployment(args) as external_deployment:
+        ports = [args.port for args in external_deployment.pod_args['pods'][0]]
         flow = Flow(name='k8s_flow-with_external_deployment', port=9090).add(
             name='external_executor',
             external=True,
-            host='external-deployment.external-deployment-ns.svc',
-            port=GrpcConnectionPool.K8S_PORT,
+            host=f'172.17.0.1',
+            port=ports[0],
```
Comment (Jackmin801) [inline tests/k8s/test_k8s.py:310]: ??

Comment (Jackmin801) [reply tests/k8s/test_k8s.py:310]: Oh it's because the diff is matching the wrong place. It matched the duplicate test_flow_with_external_k8s_deployment

**Comment by Jackmin801** [comment]
Hrmm im not sure where the codecov drop comes from. One way to go about figuring it out is to restructure this PR:

1. Have one PR that moves the k8s folder to k8s_legacy so all k8s tests are legacy
2. Port each test from legacy to the k8s folder containing new structure in separate PRs

**Comment by JoanFM** [comment]
Please add a clear description of what is being done in this change. Where the old tests are moved?

**Pull request closed without merging**